cmd/clusterctl/client/cluster/proxy_test.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/test"
+	"sigs.k8s.io/cluster-api/cmd/version"
+)
+
+var _ Proxy = &test.FakeProxy{}
+
+func TestProxyGetConfig(t *testing.T) {
+	tests := []struct {
+		name               string
+		context            string
+		currentContext     string
```

Comment:
Nope. This is a field that sets the `currentContext: ` in the kubeconfig. I specify this field to ensure that if the `currentContext` is something different from the `context` that is passed in it respects the `context` and not `currentContext`.

The assertion is indirectly happening on the host (`expectedHost`) of the corresponding cluster to the context since that's what's available from the `rest.Config` that is returned.